packages/medusa/src/loaders/defaults.ts

```diff
@@ -123,59 +127,15 @@ export default async ({
   await entityManager.transaction(async (manager: EntityManager) => {
     await storeService.withTransaction(manager).create()
 
-    const payProviders =
-      silentResolution<(typeof BasePaymentService | AbstractPaymentService)[]>(
-        container,
-        "paymentProviders",
-        logger
-      ) || []
-    const payIds = payProviders.map((p) => p.getIdentifier())
-
-    const pProviderService = container.resolve<PaymentProviderService>(
-      "paymentProviderService"
-    )
-    await pProviderService.registerInstalledProviders(payIds)
-
-    const notiProviders =
-      silentResolution<typeof BaseNotificationService[]>(
-        container,
-        "notificationProviders",
-        logger
-      ) || []
-    const notiIds = notiProviders.map((p) => p.getIdentifier())
-
-    const nProviderService = container.resolve<NotificationService>(
-      "notificationService"
-    )
-    await nProviderService.registerInstalledProviders(notiIds)
-
-    const fulfilProviders =
-      silentResolution<typeof BaseFulfillmentService[]>(
-        container,
-        "fulfillmentProviders",
-        logger
-      ) || []
-    const fulfilIds = fulfilProviders.map((p) => p.getIdentifier())
-
-    const fProviderService = container.resolve<FulfillmentProviderService>(
-      "fulfillmentProviderService"
-    )
-    await fProviderService.registerInstalledProviders(fulfilIds)
-
-    const taxProviders =
-      silentResolution<AbstractTaxService[]>(
-        container,
-        "taxProviders",
-        logger
-      ) || []
-    const taxIds = taxProviders.map((p) => p.getIdentifier())
+    const context = { container, manager, logger }
+    await registerPaymentProvider(context)
+    await registerNotificationProvider(context)
+    await registerFulfillmentProvider(context)
+    await registerTaxProvider(context)
```

Comment:
nit: can we run them in parallel?

Reply:
We can definitely run them concurrently 👍

Reply:
done @carlos-r-l-rodrigues [here](https://github.com/medusajs/medusa/pull/2978/commits/cd1b6e25d50e2b4bab018ea6ddd727bfc456ffdd)